pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java

```diff
@@ -141,7 +142,7 @@ public static List<ListenerConfig> buildServerAdminConfigs(PinotConfiguration se
     String adminApiPortString = serverConf.getProperty(CommonConstants.Server.CONFIG_OF_ADMIN_API_PORT);
     if (adminApiPortString != null) {
       listeners.add(
-          new ListenerConfig(CommonConstants.HTTP_PROTOCOL, DEFAULT_HOST, Integer.parseInt(adminApiPortString),
+          new ListenerConfig(CommonConstants.HTTP_PROTOCOL, DEFAULT_HOST, NetUtils.findOpenPort(Integer.parseInt(adminApiPortString)),
```

Comment:
this isn't meant for production because we should stick to configured port (for non test flow) rather than determine an available port. so will refactor this.